src/python/pants/option/parser.py

```diff
@@ -442,7 +442,15 @@ def to_value_type(val_str):
       elif kwargs.get('type') == bool:
         return self._ensure_bool(val_str)
       else:
-        return self._wrap_type(kwargs.get('type', str))(val_str)
+        type_arg = kwargs.get('type', str)
+        if (not kwargs.get('idempotent_type_check', True)) and isinstance(val_str, type_arg):
+          return val_str
+        else:
+          try:
+            return self._wrap_type(type_arg)(val_str)
+          except TypeError as e:
+            raise ParseError('Error applying types to option values for option {} in {}: {}'
```

Comment:
This error can mention the type and the value, no?

Reply:
Yes! Will do.

Reply:
Done, and modified testing to match!